assets/shaders/viewport_render.wgsl

```diff
@@ -548,21 +536,36 @@ fn get_by_ray(ray_: Line) -> OctreeRayIntersection{
                 target_bounds.size
             );
             node_stack[node_stack_i] = new_node_stack_item(
-                target_bounds, target_hit, target_child, child_target_octant
+                target_bounds, target_hit, target_child_key, child_target_octant
             );
             node_stack_i += 1;
         } else {
             // ADVANCE
-            let current_target_bounds = target_bounds(node_stack[node_stack_i - 1]);
-            let step_vec = dda_step_to_next_sibling(
-                ray,
-                &current_d,
-                current_target_bounds,
-                ray_scale_factors
-            );
-            node_stack[node_stack_i - 1] = add_point_to(node_stack[node_stack_i - 1], step_vec);
-            if target_hit.hit == true {
-                current_d = target_hit.exit_distance;
+            loop{
+                if ((!cube_contains_point(current_bounds, node_stack[node_stack_i - 1].child_center))
+                 || (
+                    key_might_be_valid(target_child_key)
+                    && 0u < nodes[target_child_key].contains_nodes
+                )) {
+                    break;
+                }
```

Comment:
If the current target child is out of bounds, or not empty the iteration shouldn't advance, because the next step should be either a POP or a PUSH